src/database.ts

```diff
@@ -182,13 +284,19 @@ class Database extends ServiceObject {
    *   readTimestamp: 1518464696657
    * });
    */
-  batchTransaction(identifier, options?) {
-    let session = identifier.session;
+  batchTransaction(
+    identifier: TransactionIdentifier,
+    options?: TimestampBounds
+  ): BatchTransaction {
+    const session =
+      typeof identifier.session === 'string'
+        ? this.session(identifier.session)
+        : identifier.session;
     const id = identifier.transaction;
-    if (is.string(session)) {
-      session = this.session(session);
-    }
-    const transaction = new BatchTransaction(session, options);
+    const transaction: BatchTransaction = new BatchTransaction(
+      session,
+      options
+    );
     transaction.id = id;
     transaction.readTimestamp = identifier.readTimestamp;
```

Comment:
Looks like you uncovered a bug, I think it should be `identifier.timestamp` maybe?

Reply:
it's working correctly, as transaction is type of `BatchTransaction` and identifier is type of `TransactionIdentifier`.